pdns/dnsdist-lua.cc

```diff
@@ -1572,6 +1572,57 @@ vector<std::function<void(void)>> setupLua(bool client, const std::string& confi
       }
     });
 
+    g_lua.registerFunction<void(DNSQuestion::*)(std::string, std::string)>("setTag", [](DNSQuestion& dq, const std::string& strLabel, const std::string& strValue) {
+
+      if(dq.qTag == NULL)
+        {
+         dq.qTag = std::shared_ptr<QTag>(new QTag);
```

Comment:
Replacing `std::shared_ptr<QTag>(new QTag)` by `std::make_shared<QTag>()` would save an allocation and be a bit more easy to read.